Src/HALALMock/Services/Communication/SPI/SPI.cpp

```diff
@@ -189,26 +163,11 @@ bool SPI::transmit_and_receive(uint8_t id, span<uint8_t> command_data,
         return false;
     }
 
-    SPI::Instance* spi = SPI::registered_spi[id];
-
-    HAL_StatusTypeDef errorcode =
-        HAL_SPI_TransmitReceive_DMA(spi->hspi, command_data.data(),
-                                    receive_data.data(), command_data.size());
-    turn_on_chip_select(spi);
-    switch (errorcode) {
-        case HAL_OK:
-            return true;
-            break;
-        case HAL_BUSY:
-            return false;
-            break;
-        default:
-            ErrorHandler(
-                "Error while transmiting and receiving with spi DMA. Errorcode "
-                "%u",
-                (uint8_t)errorcode);
-            return false;
-            break;
+    if (SPI::transmit(id, command_data) && SPI::receive(id, receive_data)) {
+        HAL_SPI_TxRxCpltCallback(SPI::registered_spi[id]->hspi);
```

Comment:
you are not calling a method like this one when just transmitting or receiving, so why do it here, if you want to call the HAL callbacks you should also do it in the transmit and recieve methods.

Reply:
But transmit and receive methods are empty

Reply:
Dont use the callbacks, ideally we will drop all of the HAL codebase, so we cannot be introducing dependencies. The callback basically implements a state machine using the status of the SPI bus, we should be abble to bypass it

Reply:
Transmit callback and receive callback is declared but empty, only transmission-and-reception callback is doing stuff, so I leave it as it is.